python/pyspark/ml/classification.py

```diff
@@ -2558,11 +2558,17 @@ def _from_java(cls, java_stage):
         labelCol = java_stage.getLabelCol()
         predictionCol = java_stage.getPredictionCol()
         rawPredictionCol = java_stage.getRawPredictionCol()
+        weightCol = (
+            java_stage.getWeightCol() if
+            java_stage.isDefined(java_stage.getParam("weightCol"))
+            else None)
         classifier = JavaParams._from_java(java_stage.getClassifier())
         parallelism = java_stage.getParallelism()
         py_stage = cls(featuresCol=featuresCol, labelCol=labelCol, predictionCol=predictionCol,
-                       rawPredictionCol=rawPredictionCol, classifier=classifier,
-                       parallelism=parallelism)
+                       rawPredictionCol=rawPredictionCol,
+                       classifier=classifier, parallelism=parallelism)
+        if weightCol:
+            py_stage.setWeightCol(weightCol)
```

Comment:
I think you may want to use something similar to what you have in OneVsRestModel: 
```
        if java_stage.isDefined(java_stage.getParam("weightCol"):
            py_stage.setWeightCol(java_stage.getWeightCol())
```
Then you can remove line 2561-2564. The code is simpler this way.

Reply:
Thanks @huaxingao